ReportUtilityBills/resources/util.rb

```diff
@@ -125,3 +514,113 @@ def self.calculate_monthly_prorate(header, month)
     return num_days_in_month.to_f / Constants.NumDaysInMonths(year)[month - 1]
   end
 end
+
+def valid_filename(x)
+  x = "#{x}".gsub(/[^0-9A-Za-z\s]/, '') # remove non-alphanumeric
+  x = "#{x}".gsub(/\s+/, ' ').strip # remove multiple spaces
+  return x
+end
+
+def process_usurdb(filepath)
+  require 'csv'
+  require 'json'
+
+  skip_keywords = true
+  keywords = ['lighting',
+              'lights',
+              'private light',
+              'yard light',
+              'security light',
+              'lumens',
+              'watt hps',
+              'incandescent',
+              'halide',
+              'lamps',
+              '[partial]',
+              'rider',
+              'irrigation']
+
+  puts 'Parsing CSV...'
+  rates = CSV.read(filepath, headers: true)
+  rates = rates.map { |d| d.to_hash }
+
+  puts 'Selecting residential rates...'
```

Comment:
I looked at the JSONs sorted by file size and found a few that were really small and basically have no utility rate info in them. Can you filter out tariffs that don't have the minimum fields needed to perform the calculation?

Reply:
Maybe we can add a unit test around one of these files too.

Reply:
I filtered out those without `energyweekdayschedule`, `energyweekendschedule`, or `energyratestructure`.

Reply:
I added `test_warning_missing_required_fields`.